### ISSUE TITLE
How do I use TextCleaner?

### ISSUE THREAD

**Issue by Picklefall1** [dlemstra/FredsImageMagickScripts.NET#14]
I found some documentation for textcleaner in the provided test project, and I've tried to mimic that code in my own project, but with some modifications:

![image](https://user-images.githubusercontent.com/2670953/30501866-0b27f5d6-9a32-11e7-985d-9874d124f270.png)

I'm looping through each page of a multi-page tiff, and for each page I'd like to run the textcleaner script against it. I'm guessing that I need to use TextCleanerScript.Execute() for this purpose. Now, I'm starting with a Pix object (tesseract's way of breaking up a tiff into individual pages). So I convert a Pix into a Bitmap, and then the Bitmap into a MagickImage object so I can call Execute(). Am I doing this right? I keep getting this error when I try this:

![image](https://user-images.githubusercontent.com/2670953/30502003-87a71b82-9a32-11e7-9afe-e9bc1dcb79c9.png)

Thanks for any help you can give me!

**Comment by dlemstra** [comment]
First of all you are leaking memory... and you could this a lot simpler. You can read the pages with the MagickImageCollection.
```C#
TextCleanerScript cleaner = new TextCleanerScript(): // You could do this before you load the tiff image.

string fileName = pathOnly + "\\" + fileNameOnly + ".tiff";
using (MagickImageCollection pages = new MagickImageCollection(fileName))
{
  foreach (MagickImage page in pages)
  {
    using (IMagickImage cleaned= cleaner.Execute(page))
    {
      using (Bitmap cleanedBitmap = cleaned.ToBitmap())
      {
        using (Pix pix = PixConverter.ToPix(cleanedBitmap))
        {
          // foreach(Other... etc.
        }
      }
    }
  }
}
```

I think the reason it fails is because it needs to save the pixels to disk but runs out of disk space. The pixels are saved to disk because you are running out of memory. Is your application 64-bit?

**Comment by Picklefall1** [comment]
Thanks for the fast response, dlemstra! My application is 64 bit, but I think one of the dll's is 32bit: 

![image](https://user-images.githubusercontent.com/2670953/30503299-b5c807ba-9a37-11e7-9417-fe34135dbed3.png)

That's the version of the Magick.NET dll that was provided in the project I downloaded, so I think the entire project became constrained to 32 bit because of it. Do I need to try to find a 64 bit version of this dll and use that instead?

Also, when I used your code, I got no errors, but the ocr found no text at all. I suspect that textcleaner is probably resizing the image in some way. Is it reducing it by half in both width and height?
